### PR TITLE
Clean up cart features, closes #115, closes #7

### LINKED ISSUE #7: As a guest, if I try to checkout, I get redirected to the login page and resume checkout after login

### PR DESCRIPTION
**Pull request by KristaANelson**

### CODE REVIEW

File: app/assets/stylesheets/cart_index.css.scss [hunk 1]
```diff
@@ -5,3 +5,12 @@
...
 }
+
+.checkout {
+  margin-top: 20px;
+  float: right;
+}
+
+.total_price {
+  text-align: right;
+}
```
Comment (KristaANelson) [inline app/assets/stylesheets/cart_index.css.scss:16]: styling for the Total: Price 

I feel like there is probably a better way to do this. I had to use a bunch of TD's to make it position correctly. It looks ok to the user though

File: app/controllers/application_controller.rb [hunk 2]
```diff
@@ -20,11 +20,7 @@ def redirect_after_login
     elsif session[:return_to]
       redirect_to session[:return_to]
     else
-      @order = Order.create(user_id: current_user.id, status: "ordered")
-      @order.create_order_items(@cart)
-      @order.update_attributes(total_price: @order.order_total)
-      @cart.clear
-      redirect_to order_path(@order)
+      redirect_to new_order_path
```
Comment (KristaANelson) [inline app/controllers/application_controller.rb:23]: Created an order review page before submit which was able to get rid of this hack of creating an order here.

File: app/controllers/events_controller.rb [hunk 3]
```diff
@@ -1,10 +1,11 @@
 class EventsController < ApplicationController
   def index
-    @events = Event.active
+    @items = Item.active.available.where.not(id: session[:cart])
+    @events = @items.map(&:event).uniq
```
Comment (KristaANelson) [inline app/controllers/events_controller.rb:4]: these both work but could user refactoring!

File: app/controllers/orders_controller.rb [hunk 4]
```diff
@@ -5,6 +5,7 @@ def show
 
   def new
     @order = Order.new
+    @items = Item.where(id: session[:cart])
```
Comment (KristaANelson) [inline app/controllers/orders_controller.rb:8]: I made a new order place holder for now. This works but needs some love :heart:

File: app/controllers/orders_controller.rb [hunk 5]
```diff
@@ -18,8 +19,7 @@ def index
   def create
     @order = Order.create(user_id:     current_user.id,
                           status:      "ordered")
-    @order.create_order_items(@cart)
-    @order.update_attributes(total_price: @order.order_total)
+    session[:cart] = []
```
Comment (KristaANelson) [inline app/controllers/orders_controller.rb:22]: at this point, I wasn't really worrying about creating the order details but just making sure we could get to this point.

File: app/controllers/venues_controller.rb [hunk 6]
```diff
@@ -1,7 +1,8 @@
 class VenuesController < ApplicationController
-
   def show
     @venue = Venue.find_by(id: params[:id])
-    @events = @venue.active_events
+    possible_event_ids = Event.where(venue_id: @venue.id).uniq.map(&:id)
+    @items = Item.available.active.where.not(id: session[:cart]).where(event_id: possible_event_ids)
+    @events = @items.map(&:event).uniq
```
Comment (KristaANelson) [inline app/controllers/venues_controller.rb:6]: I FINALLY got this to work but I'll need help take some of the logic out of the controller

File: app/models/cart.rb [hunk 7]
```diff
@@ -18,7 +18,7 @@ def count
   end
 
   def remove_item(item_id)
-    cart_items.except!(item_id)
+    @cart_items.delete(item_id)
```
Comment (KristaANelson) [inline app/models/cart.rb:21]: cart_items are now an array instead of a hash

File: app/models/item.rb [hunk 8]
```diff
@@ -28,7 +28,7 @@ def dollar_amount
   end
 
   def category_list
-    categories.map { |cat| cat.name }.join(", ")
+    categories.first
```
Comment (KristaANelson) [inline app/models/item.rb:31]: I think we should ditch the many relationship on categories and just have them pick one.

File: app/models/order.rb [hunk 9]
```diff
@@ -39,7 +39,7 @@ def formatted_hour
   end
 
   def total_dollar_amount
-    number_to_currency(total_price / 100.00)
+    number_to_currency(100.00 / 100.00)
```
Comment (KristaANelson) [inline app/models/order.rb:42]: place holder until we setup the create order logic

File: app/views/cart_items/index.html.erb [hunk 10]
```diff
@@ -2,7 +2,15 @@
...
+    <p> Ticket availability subject to change </p>
   </div>
   <%= render partial: "shared/cart_items" %>
+  <div class="checkout">
+  <% if current_user %>
+    <%= button_to "Checkout", orders_path(user_id: current_user.id), cart: @cart, class: 'btn btn-default btn-lg' %>
+  <% else %>
+    <%= button_to "Checkout", login_for_cart_path, method: :get, class: 'btn btn-default btn-lg'%>
+  <% end %>
+  </div>
```
Comment (KristaANelson) [inline app/views/cart_items/index.html.erb:15]: Just added a Checkout button and warning about the ticket availability

File: app/views/events/show.html.erb [hunk 11]
```diff
@@ -69,7 +69,7 @@
         </tr>
       </thead>
       <tbody>
-        <% @event.items.each do |item| %>
+        <% @items.each do |item| %>
```
Comment (KristaANelson) [inline app/views/events/show.html.erb:72]: now items only include the ones not in the cart

File: app/views/orders/new.html.erb [hunk 12]
```diff
@@ -0,0 +1,9 @@
+<div class="cart-items">
+  <h2>Review Your Order!</h2>
+</div>
+<%= render partial: "shared/cart_items" %>
+<div class="checkout">
+  <% if current_user %>
+  <%= button_to "Submit Order", orders_path(user_id: current_user.id), cart: @cart, class: 'btn btn-default btn-lg' %>
+</div>
+<% end %>
```
Comment (KristaANelson) [inline app/views/orders/new.html.erb:9]: was able to user the cart_items partial!

File: app/views/orders/show.html.erb [hunk 13]
```diff
@@ -1,5 +1,8 @@
-<h3>Order Summary</h3>
-<strong>Order placed at:</strong> <%= @order.formatted_created_at %></br>
+<div class="cart-items">
+<h3>Order Confirmation: HRAD7457MFKJ </h3>
+<p> Please check your email for details on your ticket </p>
+</div>
+<!-- <strong>Order placed at:</strong> <%= @order.formatted_created_at %></br>
```
Comment (KristaANelson) [inline app/views/orders/show.html.erb]: threw in dummy data for now until we get this implemented

File: app/views/shared/_cart_items.html.erb [hunk 14]
```diff
@@ -11,7 +11,7 @@
         <th>Section</th>
         <th>Row</th>
         <th>Seat</th>
-        <th>Delivery Method</th>
+        <th>Delivery</th>
```
Comment (KristaANelson) [inline app/views/shared/_cart_items.html.erb:14]: This was the only column that took up tow lines, so I shortened it

File: app/views/shared/_cart_items.html.erb [hunk 15]
```diff
@@ -19,9 +19,9 @@
         <% @items.each do |item| %>
         <tr class="event-row">
           <td>
-            <div class="rotate event-category">
+            <!-- <div class="rotate event-category">
               <p><%= item.event.category_name %></p>
-            </div>
+            </div> -->
```
Comment (KristaANelson) [inline app/views/shared/_cart_items.html.erb]: This was not lining up correctly. I didn't know how to fix it. We can either remove it, or we'll have to figure out how to get back where it's supposed to be

File: app/views/shared/_cart_items.html.erb [hunk 16]
```diff
@@ -38,9 +38,26 @@
           <td class="vert-align"><%= item.seat %></td>
           <td class="vert-align delivery-meth"><%= item.delivery_method %></td>
           <td class="vert-align"><%= item.dollar_amount %></td>
-          <td class="vert-align"><%= button_to  "Remove from Cart", remove_item_path(item_id: item.id), data: { confirm: 'Are you sure you want to remove this item from your cart?' }, id:"btn-danger" %></td>
+          <% if current_page?(cart_path) %>
+          <td class="vert-align"><%= button_to  "Remove", remove_item_path(item_id: item.id), data: { confirm: 'Are you sure you want to remove this item from your cart?' }, id:"btn-danger" %></td>
+          <% end %>
```
Comment (KristaANelson) [inline app/views/shared/_cart_items.html.erb:39]: they can only remove an item on the cart page, not on the order review page

File: app/views/shared/_cart_items.html.erb [hunk 17]
```diff
@@ -38,9 +38,26 @@
...
+          <td> </td>
+          <td> </td>
+          <td> </td>
+          <td> </td>
+          <td> </td>
+          <td> </td>
+          <td> <h3>Total:</h3> </td>
+        <td> <h3 class="total_price"> <%="#{(@items.map(&:unit_price).sum)/100.00}" %></h3></td>
+      </tr>
+    </tfoot>
```
Comment (KristaANelson) [inline app/views/shared/_cart_items.html.erb:57]: soooo many td's... it looks good but there has to be a better way??

**Comment by KristaANelson** [comment]
This PR turned out a lot bigger than planned! I added notes for each file to explain what was going on. functionally works, next step to refactor

File: app/models/item.rb (hunk 8 above)
Comment (scottcrawford03) [inline app/models/item.rb:31]: I agree, the many to many doesn't make sense

**Pull request merged**